src/ValueObject/Route/Cart.php

```diff
@@ -0,0 +1,33 @@
+<?php
+/**
+ * Copyright Shopgate Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @author    Shopgate Inc, 804 Congress Ave, Austin, Texas 78701 <interfaces@shopgate.com>
+ * @copyright Shopgate Inc
+ * @license   http://www.apache.org/licenses/LICENSE-2.0 Apache License, Version 2.0
+ */
+
+namespace Shopgate\CloudIntegrationSdk\ValueObject\Route;
+
+class Cart extends AbstractRoute
+{
+    public function __construct() {
+        $this->identifier    = "/carts/{$this->paramNameList[0]}";
```

Comment:
isn't it trying to access ```this->paramNameList[0]``` of a parameter that is set after it? Also, were you planning on using ```->getParamName(0)``` here?

Reply:
you're right. I had the assignment at the top, first but then thought it would look better at the botton. But I forgot I was using it...

Reply:
I've now changed the assignment order in all the files for consistency reasons.